circuit-foundation/src/commonMain/kotlin/com/slack/circuit/foundation/NavigableCircuitContent.kt

```diff
@@ -50,11 +55,30 @@ public fun NavigableCircuitContent(
     )
 
   if (backstack.size > 0) {
-    @Suppress("SpreadOperator")
     decoration.DecoratedContent(activeContentProviders, backstack.size, modifier) { provider ->
-      val values = providedValues[provider.record]?.provideValues()
-      val providedLocals = remember(values) { values?.toTypedArray() ?: emptyArray() }
-      CompositionLocalProvider(*providedLocals) { provider.content(provider.record) }
+      // We retain the record's retained state registry if the back stack
+      // contains the record
+      val record = provider.record
+      val retainChecker =
+        remember(backstack, record) { CanRetainChecker { backstack.contains(record) } }
```

Comment:
Where the magic happens. The nested registry is only retained whilst it is on the backstack.